lib/rack/directory.rb

```diff
@@ -96,6 +96,7 @@ def check_bad_request(path_info)
 
     def check_forbidden(path_info)
       return unless path_info.include? ".."
+      return if ::File.expand_path(::File.join(@root, path_info)).start_with?(@root)
```

Comment:
On windows, this will assume "\" as a path separator. Is that what you are expecting?

Reply:
It looks like it is safe to use File.join on Windows but I have no way to test it...

https://stackoverflow.com/questions/597488/how-to-do-a-safe-join-pathname-in-ruby

Reply:
This should be safe on Windows as `@root` is set to the result of `::File.expand_path`.